### PR TITLE
gtk+4 4.0.0 (new formula)

### PR DESCRIPTION
**Pull request by tschoonj**
Just one small patch necessary to compile it! That's a small Christmas miracle...

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/gtk+4.rb [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+    graphene = Formula["graphene"]
+    graphite2 = Formula["graphite2"]
+    libffi = Formula["libffi"]
+    harfbuzz = Formula["harfbuzz"]
+    libepoxy = Formula["libepoxy"]
+    libpng = Formula["libpng"]
+    pango = Formula["pango"]
+    pcre = Formula["pcre"]
+    pixman = Formula["pixman"]
+    flags = %W[
```
Comment (carlocab) [inline Formula/gtk+4.rb:87]: Can't you use `pkg-config` to generate at least some of these flags more compactly?

Comment (tschoonj) [reply Formula/gtk+4.rb:87]: Yes, and in fact this is derived from its output but there was a time when using `pkg-config` in test blocks was very much frowned upon, so this approach was necessary.

Comment (SMillerDev) [reply Formula/gtk+4.rb:87]: I can't remember that time so please use pkg-config 😁

**Comment by carlocab** [comment]
```
../modules/printbackends/gtkprintbackendcups.c:653:30: error: use of undeclared identifier 'PPD_CUSTOM_UNKNOWN'
                        case PPD_CUSTOM_UNKNOWN:
                             ^
1 error generated.
```
Only on 10.14 though. Sounds a lot like https://github.com/Homebrew/homebrew-core/pull/67035#issuecomment-747248553

**Comment by tschoonj** [comment]
Cups on Mojave is too old, so I disabled support for the printer backend on that OS and older.

**Comment by carlocab** [comment]
>Cups on Mojave is too old, so I disabled support for the printer backend on that OS and older.

Might be worth a caveat? `boost` does something like this when `clang` isn't new enough and displays a caveat for it.

**Comment by fxcoudert** [comment]
The boost caveat is antiquated and should be removed, actually :)

**Review by fxcoudert — APPROVED** [review]

**Comment by carlocab** [comment]
Ha, ok. I've got an open boost PR. I'll remove it there.

**Pull request closed without merging**